SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -198,3 +200,79 @@ bool SpatialCommandUtils::GenerateDevAuthToken(bool bIsRunningInChina, FString&
 	OutTokenSecret = TokenSecret;
 	return true;
 }
+
+bool SpatialCommandUtils::HasDevLoginTag(const FString& DeploymentName, bool bIsRunningInChina, FText& OutErrorMessage)
+{
+	if (DeploymentName.IsEmpty())
+	{
+		// If we don't specify a deployment name, we will not check if any deployment is running.
+		return true;
+	}
+
+	FString TagsCommand = FString::Printf(TEXT("project deployment tags list %s --json_output"),*DeploymentName);
+	if (bIsRunningInChina)
+	{
+		TagsCommand += SpatialGDKServicesConstants::ChinaEnvironmentArgument;
+	}
+
+	FString DeploymentCheckResult;
+	int32 ExitCode;
+	FSpatialGDKServicesModule::ExecuteAndReadOutput(*SpatialGDKServicesConstants::SpatialExe, TagsCommand, SpatialGDKServicesConstants::SpatialOSDirectory, DeploymentCheckResult, ExitCode);
+	if (ExitCode != 0)
+	{
+		FString ErrorMessage = DeploymentCheckResult;
+		TSharedRef<TJsonReader<TCHAR>> JsonReader = TJsonReaderFactory<TCHAR>::Create(DeploymentCheckResult);
+		TSharedPtr<FJsonObject> JsonRootObject;
+		if (FJsonSerializer::Deserialize(JsonReader, JsonRootObject) && JsonRootObject.IsValid())
+		{
+			JsonRootObject->TryGetStringField("error", ErrorMessage);
+		}
+		OutErrorMessage = FText::Format(LOCTEXT("DeploymentTagsRetrievalFailed", "Unable to retrieve deployment tags. Is the deployment {0} running?\nResult: {1}"), FText::FromString(DeploymentName), FText::FromString(ErrorMessage));
+		return false;
+	};
+
+	FString AuthResult;
+	FString RetrieveTagsResult;
+	bool bFoundNewline = DeploymentCheckResult.TrimEnd().Split(TEXT("\n"), &AuthResult, &RetrieveTagsResult, ESearchCase::IgnoreCase, ESearchDir::FromEnd);
+	if (!bFoundNewline || RetrieveTagsResult.IsEmpty())
+	{
+		// This is necessary because spatial might return multiple json structs depending on whether you are already authenticated against spatial and are on the latest version of it.
+		RetrieveTagsResult = DeploymentCheckResult;
+	}
+
+	TSharedRef<TJsonReader<TCHAR>> JsonReader = TJsonReaderFactory<TCHAR>::Create(RetrieveTagsResult);
+	TSharedPtr<FJsonObject> JsonRootObject;
+	if (!(FJsonSerializer::Deserialize(JsonReader, JsonRootObject) && JsonRootObject.IsValid()))
+	{
+		OutErrorMessage = FText::Format(LOCTEXT("DeploymentTagsJsonInvalid", "Unable to parse the received tags.\nResult: {0}"), FText::FromString(RetrieveTagsResult));
+		return false;
+	}
+
+
+	FString JsonMessage;
+	if (!(JsonRootObject)->TryGetStringField("msg", JsonMessage))
+	{
+		OutErrorMessage = FText::Format(LOCTEXT("DeploymentTagsMsgInvalid", "Unable to parse the msg field inside the received json data.\nResult: {0}"), FText::FromString(RetrieveTagsResult));
+		return false;
+	}
+
+	/*
+	Output looks like this:
+	Tags: [unreal_deployment_launcher,dev_login]
+	We need to parse it a bit to be able to iterate through the tags
+	*/
+	FString test = JsonMessage.Mid(7, JsonMessage.Len() - 8);
```

Comment:
Could you check that the 7th and the one before last character are `[` and `]` and error if not?
Would help catch if the format changes.